src/hooks/useNestedMenu.jsx

```diff
@@ -0,0 +1,143 @@
+import { useState, useReducer, useCallback, useEffect, createRef } from 'react';
+import Dropdown from '../components/Dropdown';
+import LinkWrapper from '../components/LinkWrapper';
+import { createReducerHelpers } from '../helpers/reducer';
+import db from '../helpers/debounce';
+import useMountDelay from './useMountDelay';
+
+const useNestedMenu = (menuItems, delay, navRef) => {
+  const [{ initState, actions, reducer }] = useState(
```

Comment:
Ok, having this hook is a bad idea to solve the task. You can notice how many times you iterate over your data structure in different handlers (map, forEach, for, etc). Besides that, you keep an array of refs. All together It makes this code really hard to read. 

I **strongly** recommend you not to use the hook here, instead create a separate component for the MenuItem which it should render the title or the nested menu, where the nested menu will recursively render.